statistics/histogram_test.go

```diff
@@ -139,3 +139,126 @@ func (s *testStatisticsSuite) TestValueToString4InvalidKey(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(res, Equals, "(1, 0.5, \x14)")
 }
+
+type bucket4Test struct {
+	lower  int64
+	upper  int64
+	count  int64
+	repeat int64
+	ndv    int64
+}
+
+func genHist4Test(buckets []*bucket4Test, totColSize int64) *Histogram {
+	h := NewHistogram(0, 0, 0, 0, types.NewFieldType(mysql.TypeLong), len(buckets), totColSize)
+	for _, bucket := range buckets {
+		lower := types.NewIntDatum(bucket.lower)
+		upper := types.NewIntDatum(bucket.upper)
+		h.AppendBucketWithNDV(&lower, &upper, bucket.count, bucket.repeat, bucket.ndv)
+	}
+	return h
+}
+
+func (s *testStatisticsSuite) TestMergePartitionLevelHist(c *C) {
+	hists := make([]*Histogram, 0, 2)
+	// Col(1) = [1, 4, 6, 9, 9, 12, 12, 12, 13, 14, 15]
+	// H(1) = [(1, 0),(4, 2),(9, 3),(12, 3),(15, 3)]
+	h1Buckets := []*bucket4Test{
+		{
+			lower:  1,
+			upper:  4,
+			count:  2,
+			repeat: 1,
+			ndv:    2,
+		},
+		{
+			lower:  6,
+			upper:  9,
+			count:  5,
```

Comment:
It seems that the meaning of the `count` in `h1Buckets` and `h2Buckets` is different from the meaning of the `count`  in `expHist`. The former is the cumulative value. Is this expected?